### PR TITLE
(chore) Expand row on edit - journal

### PR DESCRIPTION
**Pull request by mbayopanda**
This PR expand the collapsed row when editing transaction.

### CODE REVIEW

File: client/src/js/services/grid/GridGrouping.js [hunk 1]
```diff
@@ -148,6 +148,13 @@ function GridGroupingService(GridAggregators, uiGridGroupingConstants, Session,
     $timeout(api.treeBase.expandAllRows, 0, false);
   }
 
+  function unfoldGroup(row) {
+    var api = this.gridApi;
+    $timeout(function () {
```
Comment (jniles) [inline client/src/js/services/grid/GridGrouping.js]: You shouldn't need this `$timeout()` ... it's a hack and delays things getting done.  Try this:
```js
var api = this.gridApi;
api.treeBase.expandRow(row);
api.grid.notifyDataChange(uiGridConstants.dataChange.COLUMN);
```
If that works, we should also change the other `$timeout()` calls in this service.

Comment (jniles) [reply client/src/js/services/grid/GridGrouping.js]: I get the feeling `$timeout` has a negative impact on performance of this grid and we have a lot of them.

File: client/src/modules/journal/journal.js [hunk 2]
```diff
@@ -376,6 +376,10 @@ function JournalController(Journal, Sorting, Grouping,
   vm.editTransaction = editTransaction;
   function editTransaction(row) {
     vm.filterBarHeight = bhConstants.utilBar.expandedHeightStyle;
+
+    // expand the row
+    vm.grouping.unfoldGroup(row);
```
Comment (jniles) [inline client/src/modules/journal/journal.js:383]: Very  clean - and good comment!

**Review by jniles — COMMENTED** [review]
This is the correct functionality and very cleanly implemented.

`ui-grid` uses events to notify changes, and in several locations we've used a `$timeout` hack to get around limitations in `ui-grid` ... but I don't think it is necessary here.

Let me know if my solution works - if not, LGTM. 👍

File: client/src/js/services/grid/GridGrouping.js [hunk 3]
```diff
@@ -145,7 +147,14 @@ function GridGroupingService(GridAggregators, uiGridGroupingConstants, Session,
...
     api = api || this.gridApi;
-    $timeout(api.treeBase.expandAllRows, 0, false);
+    api.treeBase.expandAllRows();
+    api.grid.notifyDataChange(uiGridConstants.dataChange.COLUMN);
+  }
+
+  function unfoldGroup(row) {
+    var api = this.gridApi;
+    api.treeBase.expandRow(row);
+    api.grid.notifyDataChange(uiGridConstants.dataChange.COLUMN);
```
Comment (jniles) [inline client/src/js/services/grid/GridGrouping.js]: Beautiful :+1:

**Comment by jniles** [comment]
I'm going to wait until #1520 comes in to merge this... that way @lomamech doesn't have to change his tests yet again.

**Comment by jniles** [comment]
@kwilu r+

**Comment by ghost** [comment]
:pushpin: Commit d557d8e has been approved by `jniles`

**Comment by ghost** [comment]
:hourglass: Testing commit d557d8e4f0a6017735ac975bdcd0148e73611bc0 with merge 8ff589b4f0c94d6f648f97b0e9c427613e6da8a3...

**Comment by jniles** [comment]
@kwilu are you there?

**Comment by jniles** [comment]
@kwilu r+

**Comment by ghost** [comment]
:pushpin: Commit 4b8cdec has been approved by `jniles`

**Comment by ghost** [comment]
:hourglass: Testing commit 4b8cdec175ba34a180c6d7ea586d85cd80635406 with merge bbb94db44d0ed0c13da1ea0a33c5dbb74c51bc6b...

**Comment by jniles** [comment]
@kwilu are you still there?

**Comment by ghost** [comment]
:cake: There you are

**Comment by jniles** [comment]
@kwilu r+

**Comment by ghost** [comment]
:pushpin: Commit 8516c63 has been approved by `jniles`

**Comment by ghost** [comment]
:hourglass: Testing commit 8516c635ba136567d46146e32d2c4b9cd5ff8fb1 with merge e0c478529ecc3de61b03cf2ff0fd1d3c6ef64420...

**Pull request merged**